asynchronous apex features

Aggnieszkka opened the pull request.

Commit:
asynchronous apex features

Commit:
remaining javadocs

sylwiachr commented on InnovativeCars/force-app/main/default/classes/CarToShowroomHandler1.cls:
```diff
@@ -91,4 +91,22 @@ public class CarToShowroomHandler1 {
             }
```
Why 1 at the end of the class name?

sylwiachr commented on InnovativeCars/force-app/main/default/classes/NewOffersEmailSender.cls:
```diff
@@ -0,0 +1,33 @@
+    public static void sendEmail(List<ID> newOffersIds){
+
+        Messaging.SingleEmailMessage mail = new Messaging.SingleEmailMessage();
+        String[] toAddresses = new String[] {Constants.NEW_OFFERS_EMAIL_ADDRESS};
```
why email from constants?

sylwiachr commented on InnovativeCars/force-app/main/default/classes/DeleteOldTelemetries.cls:
```diff
@@ -0,0 +1,22 @@
+
+    public Database.QueryLocator start(Database.BatchableContext bc) {
+        date d = system.today().adddays(-90);
+        List<Telemetry__c> tel = [SELECT Id FROM Telemetry__c WHERE DAY_ONLY(CreatedDate) <= :d];
```
this line was only for the purpose of tests I guess?

sylwiachr commented on InnovativeCars/force-app/main/default/classes/DeleteOldTelemetriesTest.cls:
```diff
@@ -0,0 +1,53 @@
+    /**
+    * @author Agnieszka Ząbkowicz
+    * @date 06.2021
+    *
```
why this empty line? please also add proper date

sylwiachr commented on InnovativeCars/force-app/main/default/classes/DeleteOldTelemetriesTest.cls:
```diff
@@ -0,0 +1,53 @@
+      
+        Test.startTest();
+            DeleteOldTelemetries deleteOld = new DeleteOldTelemetries();
+        	Id batchId = Database.executeBatch(deleteOld);
```
formatting

sylwiachr commented on InnovativeCars/force-app/main/default/classes/DeleteOldTelemetriesTest.cls:
```diff
@@ -0,0 +1,53 @@
+* @author Agnieszka Ząbkowicz
+* @date 06.2021
+*
+* @description test class for DeleteOldTelemetries class
```
there should be separate class testing batch and separate one testing scheduler

sylwiachr commented on InnovativeCars/force-app/main/default/classes/Constants.cls:
```diff
@@ -7,11 +7,23 @@ public class Constants {
+    public static final String NEW_OFFERS_EMAIL_SUBJECT = 'New offers added!';
+    public static final String NEW_OFFERS_EMAIL_BODY = 'New offers have been added to showrooms. Number of offers: ';
+    public static final String CASE_SUBJECT = 'Renovation Case ';
+    public static final String CASE_ORIGIN = 'Web';
```
the name is not meaningfull. the same as case subject. please consider something like CASE_ORIGIN_WEB (similarly with case subject)

sylwiachr commented on InnovativeCars/force-app/main/default/classes/RenovationTeamRequestSender.cls:
```diff
@@ -0,0 +1,44 @@
+
+        Http server = new Http();
+        HttpRequest req = new HttpRequest();
+        req.setEndpoint(Constants.RENOVATION_ENDPOINT);
```
consider having endpoint in named credentials / custom metadata / custom setting

Commit:
Corrections from review:
CarToShowroomHandler1 => CarToShowroomHandler, ShowroomHandler1 => ShowroomHandler,
date in all javadocs,
separate classes testing batch and scheduler,
constants' naming

AdrianDlu commented on InnovativeCars/force-app/main/default/classes/DeleteOldTelemetries.cls:
```diff
@@ -0,0 +1,20 @@
+public class DeleteOldTelemetries implements Database.Batchable<sObject> {
+
+    public Database.QueryLocator start(Database.BatchableContext bc) {
+        date d = system.today().adddays(-90);
```
bad variable name d

AdrianDlu commented on InnovativeCars/force-app/main/default/classes/DeleteOldTelemetries.cls:
```diff
@@ -0,0 +1,20 @@
+        delete records;
+    }
+    public void finish(Database.BatchableContext bc){
+        System.debug(Constants.TELEMETRIES_DELETED_MESSAGE);
```
Deubugi z logging levelem

AdrianDlu commented on InnovativeCars/force-app/main/default/classes/RenovationTeamRequestSender.cls:
```diff
@@ -0,0 +1,45 @@
+            HttpRequest req = new HttpRequest();
+            req.setEndpoint(Constants.RENOVATION_ENDPOINT);
+            req.setMethod('POST');
+            req.setBody(JSON.serialize(newCases)); 
```
Do systemu remontowego mialy byc wyslane casy z Id

AdrianDlu commented on InnovativeCars/force-app/main/default/classes/TestFactory.cls:
```diff
@@ -95,4 +91,23 @@ public class TestFactory {
+    * @param number of telemetries to create and add to database
+    * @return list of created telemetries
+    */
+    public static List<Telemetry__c> createTelemetries(Integer telemetriesNumber){
```
parameter name

AdrianDlu replied:
#Ignored?

AdrianDlu commented on InnovativeCars/force-app/main/default/classes/CarToShowroomHandler.cls:
```diff
@@ -91,4 +88,21 @@ public class CarToShowroomHandler1 {
+    * and sending it to method which sends email
+    * @param newOffers list of offers objects
+    */
+    public static void sendEmailAboutNewOffers(List<Car_To_Showroom__c> newOffers) {
```
inNewOffers

Commit:
Corrections from 2nd review

Commit:
if(!Test.isRunningTest()) moved to RenovationTeamCaseCreator

AdrianDlu commented on InnovativeCars/force-app/main/default/classes/RenovationTeamCaseCreator.cls:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    public void execute(QueueableContext context){
+        if(!Test.isRunningTest()){
```
Czemu wszystko jest wylaczone z testow? mysle ze tylko linia 26 powina byc w tym ifie

Commit:
Corrections from 3rd review
